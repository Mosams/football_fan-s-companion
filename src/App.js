import { Routes, Route, BrowserRouter } from 'react-router-dom';
import Navbar from './components/Navbar';
import Details from './components/Details';
import Homepage from './components/Homepage';

function App() {
  return (
    <div className="App w-full h-screen bg-yellow-700">
      <BrowserRouter>
        <Navbar />
        <Routes>
          <Route
            path="/"
            element={
              <Homepage />
    }
          />
          <Route
            path="/details"
            element={
              <Details />
    }
          />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
