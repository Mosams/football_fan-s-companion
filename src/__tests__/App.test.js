import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';

import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import App from '../App';

let appContainer = null;
beforeEach(() => {
  appContainer = document.createElement('div');
  document.body.appendChild(appContainer);
});

afterEach(() => {
  unmountComponentAtNode(appContainer);
  appContainer.remove();
  appContainer = null;
});

describe('testing the app', () => {
  it('homepage integration', () => {
    act(() => {
      render(
        <React.StrictMode>
          <Provider store={store}>
            <App />
          </Provider>
        </React.StrictMode>, appContainer,
      );
    });

    expect(pretty(appContainer.innerHTML)).toMatchSnapshot();
  });
});
