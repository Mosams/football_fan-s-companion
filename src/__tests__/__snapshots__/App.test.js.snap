// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the app homepage integration 1`] = `
"<main class=\\"App\\">
  <nav class=\\"sticky top-0 z-50 flex flex-wrap items-center justify-between px-2 py-3 bg-gray-300 mb-3\\">
    <div class=\\"container px-4 mx-auto flex flex-wrap items-center justify-between\\">
      <div class=\\"w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start\\"><a href=\\"/\\" class=\\"flex items-center gap-4\\"><img src=\\"football.png\\" alt=\\"A football icon\\" style=\\"width: 2rem; height: 2rem;\\">
          <h1 class=\\"text-2xl font-semibold\\">Football Fan Companion</h1>
        </a><button class=\\"text-black cursor-pointer text-xl leading-none px-3 py-1 border border-solid border-transparent rounded bg-transparent block lg:hidden outline-none focus:outline-none\\" type=\\"button\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 448 512\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\">
            <path d=\\"M432 416H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-128H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-128H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-128H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z\\"></path>
          </svg></button></div>
      <div class=\\"lg:flex flex-grow items-center hidden\\" id=\\"example-navbar-danger\\" data-testid=\\"hamburger\\">
        <ul class=\\"flex flex-col lg:flex-row list-none lg:ml-auto\\">
          <li class=\\"nav-item\\"><a class=\\"px-3 py-2 flex items-center text-xs uppercase font-bold leading-snug text-black hover:opacity-75\\" href=\\"/\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 352 512\\" class=\\"text-lg leading-lg text-black opacity-75\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\">
                <path d=\\"M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z\\"></path>
              </svg></a></li>
          <li class=\\"nav-item\\"><a class=\\"px-3 py-2 flex items-center text-xs uppercase font-bold leading-snug text-black hover:opacity-75\\" href=\\"/\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 512 512\\" class=\\"fab fa-pinterest text-lg leading-lg text-black opacity-75\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\">
                <path d=\\"M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z\\"></path>
              </svg></a></li>
        </ul>
      </div>
    </div>
  </nav>
  <div class=\\"max-w-10 flex items-center justify-center mt-5 sticky top-20 z-50\\"><input type=\\"text\\" class=\\"px-2 py-2 max-w-10 h-7 border\\" placeholder=\\"Search...\\" value=\\"\\"></div>
  <ul class=\\"grid gap-3.5 m-5 lg:grid-cols-4 md:grid-cols-2 sm:grid-cols-2\\"></ul>
</main>"
`;
